# 前処理_毎勤
2026/3/17にe-Statからcsvファイルをダウンロード(表番号1)\
(https://www.e-stat.go.jp/stat-search/files?page=1&layout=datalist&toukei=00450071&tstat=000001011791&cycle=0&tclass1=000001035519&tclass2=000001144508&tclass3val=0)\
2025年12月までのデータが入っている

In [35]:
import pandas as pd
import numpy as np

In [36]:
df_maikin = pd.read_csv('../data/raw/hon-maikin-k-jissu.csv', encoding='shift-jis')

/var/folders/kr/pm1fjbp54313fwfh_dy608fh0000gn/T/ipykernel_59804/1245941545.py:1: DtypeWarning: Columns (0: 月) have mixed types. Specify dtype option on import or set low_memory=False.
  df_maikin = pd.read_csv('../data/raw/hon-maikin-k-jissu.csv', encoding='shift-jis')


## 格納データの確認

In [37]:
df_maikin.head(3)

,種別,年,月,産業分類,規模,就業形態,現金給与総額,きまって支給する給与,所定内給与,所定外給与,特別給与,総実労働時間,所定内労働時間,所定外労働時間,出勤日数,前月末労働者数,増加労働者数,減少労働者数,本月末労働者数,パートタイム労働者数
0,実数,1960,CY,C,0,0,26250,22547.0,NaN,NaN,3703.0,193.6,170.8,22.8,23.0,NaN,NaN,NaN,NaN,NaN
1,実数,1960,CY,C,4,0,28786,24682.0,NaN,NaN,NaN,193.2,168.4,24.8,22.8,NaN,NaN,NaN,NaN,NaN
2,実数,1960,CY,C,5,0,22422,19121.0,NaN,NaN,NaN,196.7,175.1,21.6,23.5,NaN,NaN,NaN,NaN,NaN


In [38]:
df_maikin.tail(3)

,種別,年,月,産業分類,規模,就業形態,現金給与総額,きまって支給する給与,所定内給与,所定外給与,特別給与,総実労働時間,所定内労働時間,所定外労働時間,出勤日数,前月末労働者数,増加労働者数,減少労働者数,本月末労働者数,パートタイム労働者数
202711,実数,2025,12,TL,T,0,632196,291279.0,270719.0,20560.0,340917.0,134.6,124.7,9.9,17.4,51865541.0,759781.0,706691.0,51918631.0,16357545.0
202712,実数,2025,12,TL,T,1,861480,373212.0,344794.0,28418.0,488268.0,160.2,146.8,13.4,19.2,35566977.0,330348.0,342744.0,35561086.0,NaN
202713,実数,2025,12,TL,T,2,132794,112822.0,109377.0,3445.0,19972.0,79.0,76.7,2.3,13.5,16298564.0,429433.0,363947.0,16357545.0,NaN


In [39]:
df_maikin.columns

Index(['種別', '年', '月', '産業分類', '規模', '就業形態', '現金給与総額', 'きまって支給する給与', '所定内給与',
       '所定外給与', '特別給与', '総実労働時間', '所定内労働時間', '所定外労働時間', '出勤日数', '前月末労働者数',
       '増加労働者数', '減少労働者数', '本月末労働者数', 'パートタイム労働者数'],
      dtype='str')

In [40]:
print('種別: ', df_maikin['種別'].unique())
print('年: ', df_maikin['年'].unique())
print('月: ', df_maikin['月'].unique())
print('産業分類: ', df_maikin['産業分類'].unique())
print('規模: ', df_maikin['規模'].unique())
print('就業形態: ', df_maikin['就業形態'].unique())

種別:  <StringArray>
['実数']
Length: 1, dtype: str
年:  [1960 1961 1962 1963 1964 1965 1966 1967 1968 1969 1970 1971 1972 1973
 1974 1975 1976 1977 1978 1979 1980 1981 1982 1983 1984 1985 1986 1987
 1988 1989 1990 1991 1992 1993 1994 1995 1996 1997 1998 1999 2000 2001
 2002 2003 2004 2005 2006 2007 2008 2009 2010 2011 2012 2013 2014 2015
 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025]
月:  ['CY' '01' '02' '03' '04' '05' '06' '07' '08' '09' '10' '11' '12' 1 2 3 4
 5 6 7 8 9 10 11 12]
産業分類:  <StringArray>
['C   ', 'D   ', 'E   ', 'E12 ', 'E13 ', 'E14 ', 'E19 ', 'E21 ', 'E22 ',
 'E23 ', 'E24 ', 'E31 ', 'F   ', 'TL  ', 'D06 ', 'D08 ', 'E18 ', 'F33 ',
 'H42 ', 'H43 ', 'H44 ', 'I-1 ', 'I56 ', 'J62 ', 'J67 ', 'M75 ', 'I57 ',
 'L71 ', 'M76 ', 'D07 ', 'E-2 ', 'R88 ', 'I51 ', 'I52 ', 'I54 ', 'E-1 ',
 'E-3 ', 'E09 ', 'E15 ', 'E28 ', 'G   ', 'G39 ', 'H   ', 'I   ', 'J   ',
 'J63 ', 'J65 ', 'K68 ', 'K70 ', 'O   ', 'O81 ', 'P   ', 'P83 ', 'P85 ',
 'Q   ', 'R89 ', 'E11 ', 'E16 ', 'E25 ', 'E26 ', 'E27 

## 文字を置き換える
種別は実数のみ→消す\
月は不揃い→揃える\
産業分類・規模・就業形態は記号表記→実際の分類名に変換\
フィルターをかけて、調査産業計のみに絞る

In [41]:
df_maikin['月'] = df_maikin['月'].replace({"CY": 99})
df_maikin['月'] = df_maikin['月'].astype(int)
df_maikin['産業分類'] = df_maikin['産業分類'].replace({
    'TL  ': '調査産業計',
    'C   ': '鉱業，採石業，砂利採取業',
    'D   ': '建設業',
    'E   ': '製造業',
    'F   ': '電気・ガス・熱供給・水道業',
    'G   ': '情報通信業',
    'H   ': '運輸業，郵便業',
    'I   ': '卸売業，小売業',
    'J   ': '金融業，保険業',
    'K   ': '不動産業，物品賃貸業',
    'L   ': '学術研究，専門・技術サービス業',
    'M   ': '宿泊業，飲食サービス業',
    'N   ': '生活関連サービス業，娯楽業',
    'O   ': '教育，学習支援業',
    'P   ': '医療，福祉',
    'Q   ': '複合サービス事業',
    'R   ': 'サービス業(他に分類されないもの)'
    })
df_maikin['規模'] = df_maikin['規模'].replace({
    '0': '30人以上',
    '4': '500人以上',
    '5': '100〜499人',
    '7': '30〜99人',
    '9': '5〜29人',
    'T': '5人以上'
    })
df_maikin['就業形態'] = df_maikin['就業形態'].replace({
    0: '形態計',
    1: '一般',
    2: 'パート'
    })

In [42]:
df_maikin = df_maikin.filter(['年', '月', '産業分類', '規模', '就業形態', '現金給与総額', 'きまって支給する給与', '所定内給与',
       '所定外給与', '特別給与', '総実労働時間', '所定内労働時間', '所定外労働時間', '出勤日数', '前月末労働者数',
       '増加労働者数', '減少労働者数', '本月末労働者数', 'パートタイム労働者数'])
df_maikin = df_maikin.query('産業分類 == "調査産業計"').reset_index(drop=True)

## 2019年を基準に指数化
base_2019に2019年平均値のdfを作り、元のdfにマージ

In [43]:
base_2019 = (
    df_maikin.query('年 == 2019 and 月 == 99')
    .filter(['産業分類', '規模', '就業形態', '現金給与総額', 'きまって支給する給与', '所定内給与', '所定外給与', '特別給与',
            '総実労働時間', '所定内労働時間', '所定外労働時間'])
    .rename(columns={
        '現金給与総額': '現金給与総額_2019',
        'きまって支給する給与': 'きまって支給する給与_2019',
        '所定内給与': '所定内給与_2019',
        '所定外給与': '所定外給与_2019',
        '特別給与': '特別給与_2019',
        '総実労働時間': '総実労働時間_2019',
        '所定内労働時間': '所定内労働時間_2019',
        '所定外労働時間': '所定外労働時間_2019'
        })
)

# 規模ごとに1行だけか確認
base_2019

,産業分類,規模,就業形態,現金給与総額_2019,きまって支給する給与_2019,所定内給与_2019,所定外給与_2019,特別給与_2019,総実労働時間_2019,所定内労働時間_2019,所定外労働時間_2019
566,調査産業計,30人以上,形態計,371408,296064.0,270847.0,25217.0,75344.0,144.4,132.0,12.4
567,調査産業計,30人以上,一般,460803,360610.0,328208.0,32402.0,100193.0,163.5,147.9,15.6
568,調査産業計,30人以上,パート,111464,108375.0,104050.0,4325.0,3089.0,89.2,86.0,3.2
569,調査産業計,500人以上,形態計,479280,367442.0,330299.0,37143.0,111838.0,149.1,134.9,14.2
570,調査産業計,500人以上,一般,545476,413006.0,370099.0,42907.0,132470.0,159.7,143.5,16.2
571,調査産業計,500人以上,パート,132545,128777.0,121828.0,6949.0,3768.0,93.6,89.5,4.1
572,調査産業計,100〜499人,形態計,373389,298611.0,273821.0,24790.0,74778.0,144.5,132.5,12.0
573,調査産業計,100〜499人,一般,457742,359449.0,328091.0,31358.0,98293.0,161.8,147.1,14.7
574,調査産業計,100〜499人,パート,117352,113949.0,109096.0,4853.0,3403.0,91.8,88.3,3.5
575,調査産業計,30〜99人,形態計,311334,255241.0,236104.0,19137.0,56093.0,141.9,130.1,11.8


In [44]:
# assertは条件が真でないときerrorを返す
assert base_2019.duplicated(['産業分類', '規模', '就業形態']).sum() == 0

In [45]:
df_maikin = df_maikin.merge(base_2019, on=['産業分類', '規模', '就業形態'],how='left')

df_maikin['現金給与総額指数'] = df_maikin['現金給与総額'] / df_maikin['現金給与総額_2019'] * 100
df_maikin['きまって支給する給与指数'] = df_maikin['きまって支給する給与'] / df_maikin['きまって支給する給与_2019'] * 100
df_maikin['所定内給与指数'] = df_maikin['所定内給与'] / df_maikin['所定内給与_2019'] * 100
df_maikin['所定外給与指数'] = df_maikin['所定外給与'] / df_maikin['所定外給与_2019'] * 100
df_maikin['特別給与指数'] = df_maikin['特別給与'] / df_maikin['特別給与_2019'] * 100
df_maikin['総実労働時間指数'] = df_maikin['総実労働時間'] / df_maikin['総実労働時間_2019'] * 100
df_maikin['所定内労働時間指数'] = df_maikin['所定内労働時間'] / df_maikin['所定内労働時間_2019'] * 100
df_maikin['所定外労働時間指数'] = df_maikin['所定外労働時間'] / df_maikin['所定外労働時間_2019'] * 100

## 毎勤の加工チェック

In [46]:
df_maikin.head()

,年,月,産業分類,規模,就業形態,現金給与総額,きまって支給する給与,所定内給与,所定外給与,特別給与,...,所定内労働時間_2019,所定外労働時間_2019,現金給与総額指数,きまって支給する給与指数,所定内給与指数,所定外給与指数,特別給与指数,総実労働時間指数,所定内労働時間指数,所定外労働時間指数
0,1970,99,調査産業計,30人以上,形態計,75670,56294.0,NaN,NaN,19376.0,...,132.0,12.4,20.373821,19.014132,NaN,NaN,25.716713,129.224377,128.712121,134.677419
1,1970,99,調査産業計,500人以上,形態計,86729,62516.0,NaN,NaN,NaN,...,134.9,14.2,18.095685,17.013842,NaN,NaN,NaN,124.346076,121.645663,150.000000
2,1970,99,調査産業計,100〜499人,形態計,75148,55818.0,NaN,NaN,NaN,...,132.5,12.0,20.125928,18.692546,NaN,NaN,NaN,128.927336,128.150943,137.500000
3,1970,99,調査産業計,30〜99人,形態計,68937,52674.0,NaN,NaN,NaN,...,130.1,11.8,22.142458,20.636967,NaN,NaN,NaN,132.346723,133.589547,118.644068
4,1971,99,調査産業計,30人以上,形態計,86834,64542.0,NaN,NaN,22292.0,...,132.0,12.4,23.379679,21.800016,NaN,NaN,29.586961,127.977839,128.636364,120.967742


In [57]:
df_maikin.columns

Index(['年', '月', '産業分類', '規模', '就業形態', '現金給与総額', 'きまって支給する給与', '所定内給与',
       '所定外給与', '特別給与', '総実労働時間', '所定内労働時間', '所定外労働時間', '出勤日数', '前月末労働者数',
       '増加労働者数', '減少労働者数', '本月末労働者数', 'パートタイム労働者数', '現金給与総額_2019',
       'きまって支給する給与_2019', '所定内給与_2019', '所定外給与_2019', '特別給与_2019',
       '総実労働時間_2019', '所定内労働時間_2019', '所定外労働時間_2019', '現金給与総額指数',
       'きまって支給する給与指数', '所定内給与指数', '所定外給与指数', '特別給与指数', '総実労働時間指数', '所定内労働時間指数',
       '所定外労働時間指数'],
      dtype='str')

In [47]:
print('年: ', df_maikin['年'].unique())
print('月: ', df_maikin['月'].unique())
print('産業分類: ', df_maikin['産業分類'].unique())
print('規模: ', df_maikin['規模'].unique())
print('就業形態: ', df_maikin['就業形態'].unique())

年:  [1970 1971 1972 1973 1974 1975 1976 1977 1978 1979 1980 1981 1982 1983
 1984 1985 1986 1987 1988 1989 1990 1991 1992 1993 1994 1995 1996 1997
 1998 1999 2000 2001 2002 2003 2004 2005 2006 2007 2008 2009 2010 2011
 2012 2013 2014 2015 2016 2017 2018 2019 2020 2021 2022 2023 2024 2025]
月:  [99  1  2  3  4  5  6  7  8  9 10 11 12]
産業分類:  <StringArray>
['調査産業計']
Length: 1, dtype: str
規模:  <StringArray>
['30人以上', '500人以上', '100〜499人', '30〜99人', '5〜29人', '5人以上']
Length: 6, dtype: str
就業形態:  ['形態計' '一般' 'パート']


## いらない列を削除
産業分類は調査産業計のみなので削除\
XXXX_2019の列（8列）を削除\
出勤日数、増加労働者数、減少労働者数の3列を削除

In [58]:
df_maikin = df_maikin.filter(['年', '月', '規模', '就業形態',
                                '現金給与総額', 'きまって支給する給与', '所定内給与', '所定外給与', '特別給与',
                                '総実労働時間', '所定内労働時間', '所定外労働時間',
                                '前月末労働者数', '本月末労働者数', 'パートタイム労働者数',
                                '現金給与総額指数', 'きまって支給する給与指数', '所定内給与指数', '所定外給与指数', '特別給与指数',
                                '総実労働時間指数', '所定内労働時間指数', '所定外労働時間指数'])

# 前処理_CPI
2026/3/17にe-Statからcsvファイルをダウンロード\
(https://www.e-stat.go.jp/stat-search/database?page=1&layout=datalist&toukei=00200573&tstat=000001150147&cycle=0&tclass1val=0)
- e-Stat上での操作
    - DBボタンを押下
    - 表示項目選択タブを選択
    - 表章項目は「指数」のみチェックを入れ、確定を押下
    - 2020年基準品目は「0163 持ち家の帰属家賃を除く総合」のみチェックを入れ、確定を押下
    - 地域は「全国」のみチェック入れ、確定を押下
    - 時間軸（年・月）はXXXX年度を除外する
        - どれでもいいので「XXXX年度」を選択
        - 同一階層の選択/解除で「解除」を押下
        - 確定を押下
    - 4項目の選定が終わったら、確定を押下
    - レイアウト設定タブを選択
    - ドラッグ＆ドロップで、「2020年基準品目」をページ上部（欄外）に、 時間軸（年・月）を行に移動
    - 設定して表示を更新を押下
    - 右上のダウンロードを押下
    - ファイル形式をCSV形式(列指向形式・Shift-JIS)に変更
    - ダウンロードを押下
    - 表ダウンロードのページに飛んだら、ダウンロードを押下
今回のファイル名はFEH_00200573_260317110412.csv

In [48]:
df_cpi = pd.read_csv('../data/raw/FEH_00200573_260317110412.csv', encoding='shift-jis')

## 格納データの確認

In [49]:
df_cpi.head(3)

,tab_code,表章項目,cat01_code,2020年基準品目,area_code,地域（2020年基準）,time_code,時間軸（年・月）,unit,value
0,1,指数,163,0163 持家の帰属家賃を除く総合,0,全国,2026000101,2026年1月,NaN,115.1
1,1,指数,163,0163 持家の帰属家賃を除く総合,0,全国,2025001212,2025年12月,NaN,115.3
2,1,指数,163,0163 持家の帰属家賃を除く総合,0,全国,2025001111,2025年11月,NaN,115.5


In [50]:
df_cpi.columns

Index(['tab_code', '表章項目', 'cat01_code', '2020年基準品目', 'area_code',
       '地域（2020年基準）', 'time_code', '時間軸（年・月）', 'unit', 'value'],
      dtype='str')

In [51]:
print('表章項目: ', df_cpi['表章項目'].unique())
print('2020年基準品目: ', df_cpi['2020年基準品目'].unique())
print('地域（2020年基準）: ', df_cpi['地域（2020年基準）'].unique())
print('時間軸（年・月）: ', df_cpi['時間軸（年・月）'].unique())

表章項目:  <StringArray>
['指数']
Length: 1, dtype: str
2020年基準品目:  <StringArray>
['0163 持家の帰属家賃を除く総合']
Length: 1, dtype: str
地域（2020年基準）:  <StringArray>
['全国']
Length: 1, dtype: str
時間軸（年・月）:  <StringArray>
[ '2026年1月', '2025年12月', '2025年11月', '2025年10月',  '2025年9月',  '2025年8月',
  '2025年7月',  '2025年6月',  '2025年5月',  '2025年4月',
 ...
  '1970年9月',  '1970年8月',  '1970年7月',  '1970年6月',  '1970年5月',  '1970年4月',
  '1970年3月',  '1970年2月',  '1970年1月',    '1970年']
Length: 729, dtype: str


## データ前処理
使う列は時間軸（年・月）とvalueの2つ\
時間軸（年・月）はXXXX年XX月（各月の値）とXXXX年（年平均値）の2つが混在しているので、「年」列と「月」列に分ける

In [52]:
df_cpi = df_cpi.filter(['時間軸（年・月）', 'value'])

#正規表現により年月分離
import re

def parse_year_month(value: str):
    """
    value: '2025年11月', '2025年' など
    return: (year, month)
    """

    # 年月（YYYY年MM月）
    m = re.fullmatch(r"(\d{4})年(\d{1,2})月", value)
    if m:
        return int(m.group(1)), int(m.group(2))

    # 年のみ（YYYY年）
    m = re.fullmatch(r"(\d{4})年", value)
    if m:
        return int(m.group(1)), 99

    # 想定外の形式
    return None, None


# 適用
df_cpi[["年", "月"]] = df_cpi["時間軸（年・月）"].apply(
    lambda x: pd.Series(parse_year_month(x))
)

In [53]:
df_cpi = df_cpi.sort_values(["年", "月"]).reset_index(drop=True)\
     .filter(['年', '月', 'value'])\
    .rename(columns={'value': 'CPI(持ち家除く総合)'})

## CPIの加工チェック

In [54]:
df_cpi.head(3)

,年,月,CPI(持ち家除く総合)
0,1970,1,30.5
1,1970,2,30.7
2,1970,3,30.9


# 2つのデータフレームを合体

In [59]:
df = pd.merge(df_maikin, df_cpi, on=['年', '月'], how='left')

In [60]:
df.head()

,年,月,規模,就業形態,現金給与総額,きまって支給する給与,所定内給与,所定外給与,特別給与,総実労働時間,...,パートタイム労働者数,現金給与総額指数,きまって支給する給与指数,所定内給与指数,所定外給与指数,特別給与指数,総実労働時間指数,所定内労働時間指数,所定外労働時間指数,CPI(持ち家除く総合)
0,1970,99,30人以上,形態計,75670,56294.0,NaN,NaN,19376.0,186.6,...,NaN,20.373821,19.014132,NaN,NaN,25.716713,129.224377,128.712121,134.677419,31.3
1,1970,99,500人以上,形態計,86729,62516.0,NaN,NaN,NaN,185.4,...,NaN,18.095685,17.013842,NaN,NaN,NaN,124.346076,121.645663,150.000000,31.3
2,1970,99,100〜499人,形態計,75148,55818.0,NaN,NaN,NaN,186.3,...,NaN,20.125928,18.692546,NaN,NaN,NaN,128.927336,128.150943,137.500000,31.3
3,1970,99,30〜99人,形態計,68937,52674.0,NaN,NaN,NaN,187.8,...,NaN,22.142458,20.636967,NaN,NaN,NaN,132.346723,133.589547,118.644068,31.3
4,1971,99,30人以上,形態計,86834,64542.0,NaN,NaN,22292.0,184.8,...,NaN,23.379679,21.800016,NaN,NaN,29.586961,127.977839,128.636364,120.967742,33.1


In [61]:
df.columns

Index(['年', '月', '規模', '就業形態', '現金給与総額', 'きまって支給する給与', '所定内給与', '所定外給与',
       '特別給与', '総実労働時間', '所定内労働時間', '所定外労働時間', '前月末労働者数', '本月末労働者数',
       'パートタイム労働者数', '現金給与総額指数', 'きまって支給する給与指数', '所定内給与指数', '所定外給与指数',
       '特別給与指数', '総実労働時間指数', '所定内労働時間指数', '所定外労働時間指数', 'CPI(持ち家除く総合)'],
      dtype='str')

# 各種給与およびその指数を実質化

In [62]:
df = df.assign(実質_現金給与総額指数 = df['現金給与総額指数']/df['CPI(持ち家除く総合)']*100,
            実質_きまって支給する給与指数 = df['きまって支給する給与指数']/df['CPI(持ち家除く総合)']*100,
            実質_所定内給与指数 = df['所定内給与指数']/df['CPI(持ち家除く総合)']*100,
            実質_所定外給与指数 = df['所定外給与指数']/df['CPI(持ち家除く総合)']*100,
            実質_特別給与指数 = df['特別給与指数']/df['CPI(持ち家除く総合)']*100,
            実質_現金給与総額 = df['現金給与総額']/df['CPI(持ち家除く総合)']*100,
            実質_きまって支給する給与 = df['きまって支給する給与']/df['CPI(持ち家除く総合)']*100,
            実質_所定内給与 = df['所定内給与']/df['CPI(持ち家除く総合)']*100,
            実質_所定外給与 = df['所定外給与']/df['CPI(持ち家除く総合)']*100,
            実質_特別給与 = df['特別給与']/df['CPI(持ち家除く総合)']*100)

In [63]:
df.head()

,年,月,規模,就業形態,現金給与総額,きまって支給する給与,所定内給与,所定外給与,特別給与,総実労働時間,...,実質_現金給与総額指数,実質_きまって支給する給与指数,実質_所定内給与指数,実質_所定外給与指数,実質_特別給与指数,実質_現金給与総額,実質_きまって支給する給与,実質_所定内給与,実質_所定外給与,実質_特別給与
0,1970,99,30人以上,形態計,75670,56294.0,NaN,NaN,19376.0,186.6,...,65.092079,60.748026,NaN,NaN,82.162021,241757.188498,179853.035144,NaN,NaN,61904.153355
1,1970,99,500人以上,形態計,86729,62516.0,NaN,NaN,NaN,185.4,...,57.813691,54.357322,NaN,NaN,NaN,277089.456869,199731.629393,NaN,NaN,NaN
2,1970,99,100〜499人,形態計,75148,55818.0,NaN,NaN,NaN,186.3,...,64.300088,59.720596,NaN,NaN,NaN,240089.456869,178332.268371,NaN,NaN,NaN
3,1970,99,30〜99人,形態計,68937,52674.0,NaN,NaN,NaN,187.8,...,70.742677,65.932801,NaN,NaN,NaN,220246.006390,168287.539936,NaN,NaN,NaN
4,1971,99,30人以上,形態計,86834,64542.0,NaN,NaN,22292.0,184.8,...,70.633473,65.861076,NaN,NaN,89.386590,262338.368580,194990.936556,NaN,NaN,67347.432024


# 処理したデータフレームをcsvで出力

In [64]:
df.to_csv('../data/processed/毎勤CPI処理_調査産業計_20260317.csv', index=False, encoding='utf-8-sig')